### PR TITLE
Cr228 remove dto level validation for postal fulfilment request

### PR DESCRIPTION
**Pull request by PeterBochel**
# Motivation and Context
This modification moves some 'notNull' validation from a request object into the code. The validation of contact details (title, forename and surname) and now only done for individual cases.

# What has changed
Core code has been modified and test cases have been extended to add 2 new test methods: 1 for testing that a case for an individual fails if any of the contact fields (title, forename or surname) are empty, and another test for verifying that HH and CE cases will allow empty contact fields.

# How to test?
Run all tests.

# Links
See: https://collaborate2.ons.gov.uk/jira/browse/CR-228

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java [hunk 1]
```diff
@@ -210,6 +208,20 @@ private FulfilmentRequestedEvent createFulfilmentEvent(
...
+    boolean requestIsForIndividualProduct =
+        product.getCaseType() == Product.CaseType.HI
+            || product.getCaseType() == Product.CaseType.CI;
+    if (requestIsForIndividualProduct) {
+      if (StringUtils.isBlank(contact.getTitle())
+          || StringUtils.isBlank(contact.getForename())
+          || StringUtils.isBlank(contact.getSurname())) {
+        throw new CTPException(
+            Fault.BAD_REQUEST,
+            "The case is for an individual so none of the following fields can be empty: "
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java]: The fulfilment is for an individual

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java [hunk 2]
```diff
@@ -210,6 +208,20 @@ private FulfilmentRequestedEvent createFulfilmentEvent(
         Region.valueOf(caseServiceClient.getCaseById(caseId, false).getRegion().substring(0, 1));
     Product product = findProduct(fulfilmentCode, deliveryChannel, region);
 
+    boolean requestIsForIndividualProduct =
+        product.getCaseType() == Product.CaseType.HI
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java]: better : if (!isHouseholdOrCommunal)

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java (hunk 1 above)
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java]: Error text updated

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java (hunk 2 above)
Comment (PeterBochel) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java]: Now using the local method for deciding if household or communal
